File: dixit-companion/src/main/java/fr/erban/dxitcompanion/game/turn/adapter/PlayerSelectionAdapter.java
package fr.erban.dxitcompanion.game.turn.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.CheckBox;
import android.widget.TextView;

import java.util.List;

import fr.erban.dxitcompanion.R;
import fr.erban.dxitcompanion.game.turn.SelectPlayerRow;

public class PlayerSelectionAdapter extends BaseAdapter {

    private final Context mContext;

    private LayoutInflater inflater;

    private final List<SelectPlayerRow> itemsItems;

    public PlayerSelectionAdapter(Context context, List<SelectPlayerRow> itemsItems) {

        this.mContext = context;
        this.itemsItems = itemsItems;

    }

    @Override
    public int getCount() {

        return itemsItems.size();
    }

    @Override
    public Object getItem(int location) {

        return itemsItems.get(location);
    }

    @Override
    public long getItemId(int position) {

        return position;
    }

    @Override
    public View getView(int position, View playerSelectionView, ViewGroup parent) {

        ViewHolder holder;
        if (inflater == null) {
            inflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        }
        if (playerSelectionView == null) {

            assert inflater != null;
            playerSelectionView = inflater.inflate(R.layout.player_select_row, parent, false);
            holder = new ViewHolder();
            holder.name = playerSelectionView.findViewById(R.id.name);
            holder.checked = playerSelectionView.findViewById(R.id.checked);

            playerSelectionView.setTag(holder);

        } else {
            holder = (ViewHolder) playerSelectionView.getTag();
        }

        final SelectPlayerRow m = itemsItems.get(position);
        holder.name.setText(m.getName());
        holder.checked.setChecked(m.isChecked());

        return playerSelectionView;
    }

    static class ViewHolder {

        TextView name;

        CheckBox checked;

    }

}

File: dixit-companion/src/main/java/fr/erban/dxitcompanion/game/turn/adapter/PlayerSelectionAdapter.kt
package fr.erban.dxitcompanion.game.turn.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import fr.erban.dxitcompanion.databinding.PlayerSelectRowBinding
import fr.erban.dxitcompanion.game.turn.SelectPlayerRow

class PlayerSelectionAdapter(
    context: Context,
    private val items: List<SelectPlayerRow>
) : BaseAdapter() {

    private val inflater = LayoutInflater.from(context)

    override fun getCount() = items.size
    override fun getItem(position: Int): Any = items[position]
    override fun getItemId(position: Int) = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val binding = if (convertView == null) {
            PlayerSelectRowBinding.inflate(inflater, parent, false).also { it.root.tag = it }
        } else {
            convertView.tag as PlayerSelectRowBinding
        }
        val item = items[position]
        binding.name.text = item.name
        binding.checked.isChecked = item.checked
        return binding.root
    }
}
